icu4j/build.xml

```diff
@@ -1308,7 +1292,11 @@
         <ant dir="${icu4j.samples.dir}" inheritAll="false"/>
     </target>
 
-    <target name="build-tools" description="Build build-tool classes">
+    <target name="_checkBuildToolsSupported" unless="is.java8">
+        <fail>JDK8 is required to build build-tools.</fail>
+    </target>
+
+    <target name="build-tools" depends="_checkBuildToolsSupported" description="Build build-tool classes">
```

Comment:
We depends on unstable javadoc taglet APIs. Our current source works with JDK 8, but not compatible with newer JDK version. I added a check and a build target requiring javadoc tooling will be failed when JDK 9 or newer version is used.

Reply:
I think I want to create another PR for further clean up. There are some targets not working at this point. Also some other target seems probably no use/not really maintained. I will add some more comments at that time.